src/loaders/ObjectLoader.js

```diff
@@ -142,9 +138,18 @@ Object.assign( ObjectLoader.prototype, {
 
 		}
 
-		if ( json.images === undefined || json.images.length === 0 ) {
+		// fix #12601
+		// Wrap LoadingManager onLoad methods with ObjectLoader onLoad callback,
+		// allow proper callback call order
+		if ( onLoad !== undefined ) {
+
+			var loadingManagerOnLoad = scope.manager.onLoad;
+			scope.manager.onLoad = function warpedOnLoad() {
```

Comment:
Hmm, everything looks great except this part...
The only "pretty" solution I can think of is making `LoadingManager` extend `EventDispatcher`.

Reply:
Yes this is the point of interest of this PR...
Extend LoadingManager with EventDispatcher will allow to listen event like:

```javascript
if ( onLoad !== undefined ) {

	scope.manager.addEventListener( 'onLoad', function () {

		onLoad( object );

	} );

}
```

with inside LoadingManager things like:

```javascript
this.itemEnd = function ( url ) {

	itemsLoaded ++;

	if ( scope.onProgress !== undefined ) {

		scope.dispatchEvent( { type: 'onProgress' } );		// Note here
		scope.onProgress( url, itemsLoaded, itemsTotal );

	}

	if ( itemsLoaded === itemsTotal ) {

		isLoading = false;

		if ( scope.onLoad !== undefined ) {

			scope.dispatchEvent( { type: 'onLoad' } );	// Note here
			scope.onLoad();

		}

	}

};
```

Or maybe events should be called: beforeOnProgress, beforeOnLoad, beforeOnError which are more just about the underlying meaning...

Reply:
Should i make an other PR about the LoadingManager extends ?